rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs

```diff
@@ -723,3 +723,96 @@ fn test_list_known_neurons() {
+                    // A hundred million ICP should be enough
+                    icp_ledger::Tokens::from_tokens(100_000_000).unwrap(),
+                )]
+                .into_iter()
+                .collect(),
+            )
+            .with_governance_canister()
+            .build()
+            .await;
+
+        // Stake the minimum amount 100 million e8s
+        let staked_amount = 100_000_000u64;
+        let neuron_index = 0;
+
+        env.rosetta_client
+            .create_neuron(
+                env.network_identifier.clone(),
+                &(*TEST_IDENTITY).clone(),
+                RosettaCreateNeuronArgs::builder(staked_amount.into())
+                    .with_neuron_index(neuron_index)
+                    .build(),
+            )
+            .await
+            .unwrap();
+
+        // See if the neuron was created successfully
+        let agent = get_test_agent(env.pocket_ic.url().unwrap().port().unwrap()).await;
+        let neuron = list_neurons(&agent).await.full_neurons[0].to_owned();
+
+        let neuron_info = NeuronResponse::try_from(
+            TransactionOperationResults::try_from(
+                env.rosetta_client
+                    .get_neuron_info(
+                        env.network_identifier.clone(),
+                        RosettaNeuronInfoArgs::builder(neuron_index)
+                            .with_principal_id(TEST_IDENTITY.sender().unwrap().into())
+                            .build(),
+                        &(*TEST_IDENTITY).clone(),
+                    )
+                    .await
+                    .unwrap()
+                    .metadata,
+            )
+            .unwrap()
+            .operations
+            .first()
+            .unwrap()
+            .clone()
+            .metadata
+            .unwrap(),
+        )
+        .unwrap();
+
+        assert_eq!(neuron_info.neuron_id, neuron.id.unwrap().id);
+        assert_eq!(neuron_info.controller.0, TEST_IDENTITY.sender().unwrap());
+
+        let neuron_info = NeuronResponse::try_from(
+            TransactionOperationResults::try_from(
+                env.rosetta_client
+                    .get_neuron_info(
+                        env.network_identifier.clone(),
+                        RosettaNeuronInfoArgs::builder(neuron_index)
+                            .with_public_key((&Arc::new(test_identity())).into())
+                            .build(),
+                        &(*TEST_IDENTITY).clone(),
+                    )
+                    .await
+                    .unwrap()
+                    .metadata,
+            )
+            .unwrap()
+            .operations
+            .first()
+            .unwrap()
+            .clone()
+            .metadata
+            .unwrap(),
+        )
+        .unwrap();
+
+        assert_eq!(neuron_info.neuron_id, neuron.id.unwrap().id);
+        assert_eq!(neuron_info.controller.0, TEST_IDENTITY.sender().unwrap());
+    });
```

Comment:
Should we add a test for trying to get a neuron that doesn't exist?

Reply:
Done